src/core/node/elementNode.ts

```diff
@@ -457,12 +460,8 @@ export class ElementNode extends Object {
       log('Layout: ', this);
       let changedLayout = false;
       if (isFunc(this.onBeforeLayout)) {
-        changedLayout = this.onBeforeLayout.call(
-          this,
-          this,
-          child,
-          dimensions,
-        ) as boolean;
+        changedLayout =
+          this.onBeforeLayout.call(this, this, child, dimensions) ?? false;
```

Comment:
im not sure if this is a my app problem, but when i try to build this into solid-ui i get the following error here:

```
../../../../solid/src/core/node/elementNode.ts:463:9 - error TS2322: Type 'boolean | void' is not assignable to type 'boolean'.
  Type 'void' is not assignable to type 'boolean'.
  ```
  
  it seems like `??` should guard against this but the compiler still doesn't like it

Reply:
I've just tried to build solid-ui with local solid folder and don't see this error

Reply:
I got it running tsc

Reply:
I relplaced ?? with ||

Reply:
ugh that's my bad, i should've been clear, i was testing in a branch where I'm converting to tsc 😅 thanks! cleared it up for me as well